lib/cosmos/tools/limits_monitor/limits_monitor.rb

```diff
@@ -350,11 +358,14 @@ def reset
       @items = {}
       @out_of_limits = []
       @stale = []
-      unsubscribe_limits_events(@queue_id) if @queue_id
-      set_replay_mode(!get_replay_mode()) if @toggle_mode
-      @toggle_mode = false
+      if @toggle_mode
+        set_replay_mode(!get_replay_mode())
+        @clear_log_callback.call
+        @toggle_mode = false
+      end
       @clear_items_callback.call
 
+      unsubscribe_limits_events(@queue_id) if @queue_id
```

Comment:
I don't get this change.   Don't you need to unsubscribe before toggling modes, to be sure to unsubscribe from the correct server?

Reply:
Yep, I think this was a left over from my experimentation. I fixed it.